addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -2523,6 +2523,57 @@ options.registry.InnerChart = options.Class.extend({
     },
 });
 
+options.registry.ScrollButton = options.Class.extend({
+    /**
+     * @override
+     */
+    start: function () {
+        const $button = this.$('.o_scroll_button');
+        if ($button.length) {
+            this.button = $button[0];
+        } else {
+            this.button = document.createElement('button');
+            this.button.classList.add('o_scroll_button', 'fa', 'fa-angle-down', 'd-none');
+            this.button.title = _t("Scroll down to next section");
+            this.$target.append(this.button);
+        }
+    },
+    /**
+     * @override
+     */
+    cleanForSave: function () {
+        if (this.button && [...this.button.classList].includes('d-none')) {
+            this.button.remove();
+        }
+    },
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    /**
+     * Toggles the scroll down button.
+     */
+    toggleButton: function (previewMode, widgetValue, params) {
+        this.button.classList.toggle('d-none', widgetValue !== "true");
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    _computeWidgetState: function (methodName, params) {
+        switch (methodName) {
+            case 'toggleButton':
+                return ![...this.button.classList].includes('d-none');
+        }
+        return this._super(...arguments);
+    },
+});
```

Comment:
I understand your logic but it would be less code to just build and destroy the button in the option method, and less code is better ;) (no cleanForSave, simpler start, simpler _computeWidgetState)

Reply:
It was mostly for color permanence (if you toggle the option on and off, the color gets reset) but it's probably not worth how much code it adds.